torch/ao/quantization/fx/_model_report/_detector.py

```diff
@@ -134,9 +119,151 @@ def _detect_per_channel_helper(model: nn.Module):
             )
 
     if optimizations_possible:
-        further_optims_str += "To use per_channel quantization, make sure the qconfig has a per_channel weight observer."
+        further_optims_str += (
+            "To use per_channel quantization, make sure the qconfig has a per_channel weight observer."
+        )
     else:
         further_optims_str += "No further per_channel optimizations possible."
 
     # return the string and the dictionary form of same information
     return (further_optims_str, per_channel_info)
+
+
+# names for the pre and post observers that are inserted
+DEFAULT_PRE_OBSERVER_NAME = "model_report_pre_observer"
+DEFAULT_POST_OBSERVER_NAME = "model_report_post_observer"
+
+# naming conventions for stationary vs non-stationary data
+DEFAULT_STATIONARY = "stationary"
+DEFAULT_NON_STATIONARY = "non-stationary"
+
+
+def _detect_dynamic_vs_static(model: GraphModule, tolerance=0.5) -> Tuple[str, Dict[str, Any]]:
+    """
+    determines whether dynamic or static quantization is more appropriate for a given module
+
+    Stationary distribution of data are strictly above tolerance level for the comparison statistic:
+
+        S = average_batch_activation_range/epoch_activation_range
+
+    Nonstationary distributions are below the tolerance level for this metric
+
+    This will then generate suggestions for dynamic vs static quantization focused around Linear
```

Comment:
can we write a summary of the logic of how we recommend the quantization mode here?